draft-ietf-tsvwg-multipath-dccp.mkd

```diff
@@ -1013,20 +1013,38 @@ handshake is as follows:
 
 When a subflow fails to operate within the MP-DCCP requirements, it is 
 necessary to fall back to the safe operation. This may be either falling back 
-to regular DCCP, or removing a problematic subflow. The main reason for 
-subflow failing is loss of MP-DCCP options. 
+to regular DCCP, or removing a problematic subflow. The main reasons for 
+subflow failing include: loss of MP-DCCP options, failure to negotiate protocol 
+version, or modification of payload data. 
 
 At the start of the MP-DCCP connection, the handshake ensures exchange of MP-DCCP
 options and thus ensures that the path is fully MP-DCCP capable. If during the
 handshake procedure it appears that DCCP-Request or DCCP-Response or DCCP-Ack
 messages don’t have the MP-DCCP options, the MP-DCCP connection will not be 
```

Comment:
I think the distinction between DCCP feature and MP-DCCP options has to be clear here as well, MP_CAPABLE is defined as a DCCP feature not as MP-DCCP option, the only MP-DCCP option carried in the initial Request, Response and Ack messages is MP_KEY. Based on that, I think the fallback to DCCP should only happen when the MP_CAPABLE feature is not present.
```suggestion
handshake procedure it appears that the DCCP-Request or the DCCP-Response messages don't carry the MP_CAPABLE feature, the MP-DCCP connection will not be 
```

Reply:
@NathalieRM 
The difference between feature and option I already tried to fix in my suggestion [above](https://github.com/markusa/ietf-multipath-dccp/pull/78#discussion_r901595200).

I don't tend to agree that we can limit the fallback to the feature negotiation. What happens if the request Key Type is not supported or if the final ACK echoing the wrong Key A/B? What happens if the wrong Path token is used in MP_JOIN? ...

Reply:
You are right, it shouldn't happen only if the MP_CAPABLE feature is not present, the point is the text stated initially that the fallback should happen due to the abscence of MP-DCCP options in the initial 3 messages,  and I think it should happen due to the abscence of the MP_CAPABLE feature. The scenarios you mention are not abscence of MP options, and they could have different ways of handling, for the case of not supported key type and echoing wrong key, fallling back to DCCP is fine. For the case of wrong path token in MP_JOIN, I think the establishment of the subflow should just fail, at that point the initial subflow has already been established and the connection is already an MP-DCCP connection, so I don't see the reason to fallback to DCCP

Reply:
D'accord

Reply:
Nathalie's comment is not conclusive - 'shouldn't happen only' I read as 'should happen for other reasons, too', although the rest of her comment agrees with Markus's edit above, and does not provide other reasons for fallback. I have therefore made the change according to Markus's edit.

Reply:
Veselin, I think we also need some considerations what happen if MP Options are missing or faulty as outlined in this thread.

Reply:
Solved in https://github.com/markusa/ietf-multipath-dccp/pull/78/commits/3d0b91dc404d079acf13a9bd8fde0bed1848a576